crates/database/src/byte_array.rs

```diff
@@ -9,6 +9,12 @@ use sqlx::{
 #[derive(Clone, Copy, Debug)]
 pub struct ByteArray<const N: usize>(pub [u8; N]);
 
+impl<const N: usize> std::hash::Hash for ByteArray<N> {
```

Comment:
Clippy complained: It says that one should not have a derived Hash without a derived PartialEq implementation. Because then there is no guarantee that they are compatible.
(But I could also just deactivate the clippy warning, if you prefer)

Reply:
You can try deriving the PartialEq and Eq impls too. I'm not sure why I manually implemented them. It might be the case that you still can't derive them with const generic sized arrays.